remove reference to opentelemetry-erlang-api in the README (#172)
open-telemetry/opentelemetry-erlang: README.md

The [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification) defines a language library as having 2 components, the API and the SDK. The API must not only define the interfaces of any implementation in that language but also be able to function as a noop implementation of the tracer. The SDK is the default implementation of the API that must be optional.

> If you are instrumenting a project your application should only depend on the

Reviewer: When would I be instrumenting a "project"?

Should this be "if you are instrumenting a project that others will depend upon"?

Author: Even if the application being instrumented isn't used by others the user should still stick to the API unless they are writing something that requires direct SDK access, like an exporter.

Reviewer: Can i suggest "if you are instrumenting" becomes "when instrumenting"

Author: Yea, that sounds better. Can you make a github suggestion so I can just merge it in here.

Reviewer: ```suggestion
when instrumenting a project your application should only depend on the
```
